Libraries/dotNetRdf.LinkedDataFragments/Parameters.cs

```diff
@@ -0,0 +1,102 @@
+﻿/*
+// <copyright>
+// dotNetRDF is free and open source software licensed under the MIT License
+// -------------------------------------------------------------------------
+// 
+// Copyright (c) 2009-2023 dotNetRDF Project (http://dotnetrdf.org/)
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is furnished
+// to do so, subject to the following conditions:
+// 
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR 
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, 
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+// WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+// CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// </copyright>
+*/
+
+using System;
+using System.Collections.Generic;
+using Resta.UriTemplates;
+using VDS.RDF.LinkedPatternFragments.Hydra;
+using VDS.RDF.Writing.Formatting;
+
+namespace VDS.RDF.LinkedPatternFragments
+{
+    internal class Parameters
+    {
+        private readonly IriTemplate template;
+        private readonly INode subject;
+        private readonly INode predicate;
+        private readonly INode @object;
+
+        internal Parameters(IriTemplate template, INode subject = null, INode predicate = null, INode @object = null)
```

Comment:
Could we change the signature of this constructor to use IUriNode arguments. That would make the NodeType checks redundant and provide a more type-safe interface.

Reply:
I don't think that would be better.

This constructor practically takes a (possibly truncated) triple, and our `VDS.RDF.Triple` allows all node types in all positions.
The `subject` and `predicate` parameters here come from places like `Triple.Subject` and `BaseTripleCollection.WithSubject` (etc.), all of which are `INode`s.

I think the generality + checks here are the best compromise.

Reply:
That's a really good point. I hadn't taken that into consideration in my initial review.